### PR TITLE
[New Rule] Safari Settings Modification using Defaults Command

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/860

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/defense_evasion_safari_config_change.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+description = """
+Identifies changes to Safari configuration using the built-in defaults command. Adversaries may attempt to enable or
+disable certain Safari settings such as enabling JavaScript from Apple Events or equivalent to ease users browsing
+hijack.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Safari Settings Modification using Defaults Command"
```
Comment (bm11100) [inline rules/macos/defense_evasion_safari_config_change.toml:17]:
```suggestion
name = "Modification of Safari Settings via Defaults Command"
```

Comment (bm11100) [reply rules/macos/defense_evasion_safari_config_change.toml:17]: for consistency with other rule names

File: rules/macos/defense_evasion_safari_config_change.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+creation_date = "2021/01/14"
+maturity = "production"
+updated_date = "2021/01/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies changes to Safari configuration using the built-in defaults command. Adversaries may attempt to enable or
+disable certain Safari settings such as enabling JavaScript from Apple Events or equivalent to ease users browsing
+hijack.
```
Comment (bm11100) [inline rules/macos/defense_evasion_safari_config_change.toml:11]:
```suggestion
Identifies changes to the Safari configuration using the built-in defaults command. Adversaries may attempt to enable or
disable certain Safari settings, such as enabling JavaScript from Apple Events to ease in the hijacking of the users browser.
```

Comment (bm11100) [reply rules/macos/defense_evasion_safari_config_change.toml:11]: re-lint

File: rules/macos/defense_evasion_safari_config_change.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Safari Settings Modification using Defaults Command"
+references = ["https://objectivebythesea.com/v2/talks/OBTS_v2_Zohar.pdf"]
+risk_score = 43
```
Comment (bm11100) [inline rules/macos/defense_evasion_safari_config_change.toml:19]:
```suggestion
risk_score = 47
```

File: rules/macos/defense_evasion_safari_config_change.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+query = '''
+event.category:process and event.type:(start or process_started) and
+ process.name:defaults and process.args:
+                               (com.apple.Safari and write and 
+                                not (UniversalSearchEnabled or 
+                                     SuppressSearchSuggestions or 
+                                     WebKitTabToLinksPreferenceKey or 
+                                     ShowFullURLInSmartSearchField or 
+                                     com.apple.Safari.ContentPageGroupIdentifier.WebKit2TabsToLinks)
+                               )
```
Comment (bm11100) [inline rules/macos/defense_evasion_safari_config_change.toml:34]:
```suggestion
event.category:process and event.type:(start or process_started) and
  process.name:defaults and process.args:
    (com.apple.Safari and write and not
      (
      UniversalSearchEnabled or
      SuppressSearchSuggestions or
      WebKitTabToLinksPreferenceKey or
      ShowFullURLInSmartSearchField or
      com.apple.Safari.ContentPageGroupIdentifier.WebKit2TabsToLinks
      )
    )
```

Comment (bm11100) [reply rules/macos/defense_evasion_safari_config_change.toml:34]: query spacing and a bit more readable like below:

```
event.category:process and event.type:(start or process_started) and
  process.name:defaults and process.args:
    (com.apple.Safari and write and not
      (
      UniversalSearchEnabled or
      SuppressSearchSuggestions or
      WebKitTabToLinksPreferenceKey or
      ShowFullURLInSmartSearchField or
      com.apple.Safari.ContentPageGroupIdentifier.WebKit2TabsToLinks
      )
    )
```

File: rules/macos/defense_evasion_safari_config_change.toml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+name = "Safari Settings Modification using Defaults Command"
+references = ["https://objectivebythesea.com/v2/talks/OBTS_v2_Zohar.pdf"]
+risk_score = 43
+rule_id = "6482255d-f468-45ea-a5b3-d3a7de1331ae"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Defense Evasion"]
+type = "query"
+
+query = '''
+event.category:process and event.type:(start or process_started) and
```
Comment (peasead) [inline rules/macos/defense_evasion_safari_config_change.toml:26]:
```suggestion
event.category:process and event.type:start and
```
`process_started` isn't used anymore for `auditbeat`

**Review by threat-punter — APPROVED** [review]
👍

Can you add these PRs to the `detection engine rules release` project for tracking?

**Review by bm11100 — APPROVED** [review]

**Pull request merged**